projects/igniteui-angular/src/lib/checkbox/checkbox.component.ts

```diff
@@ -492,29 +493,43 @@ export class IgxCheckboxComponent implements EditorProvider, AfterViewInit, Cont
      * @internal
      */
     private _required = false;
+    private _prefersTokenizedTheme = false;
     private elRef = inject(ElementRef);
-    private _theme$ = new Subject<IgxTheme>();
-    private _subscription: Subscription;
     private destroyRef = inject(DestroyRef);
 
     constructor(
         protected cdr: ChangeDetectorRef,
         protected renderer: Renderer2,
-        protected themeService: ThemeService,
+        @Inject(THEME_TOKEN)
+        protected themeToken: ThemeToken,
         @Optional() @Self() public ngControl: NgControl,
     ) {
         if (this.ngControl !== null) {
             this.ngControl.valueAccessor = this;
         }
 
-        this.theme = this.themeService.globalTheme;
+        this.theme = this.themeToken.theme;
+        this._prefersTokenizedTheme = this.themeToken.preferToken;
```

Comment:
Very minor: You already have the whole token locally and can possibly skip another local prop for the prefer flag. That said `theme` can also be a simple getter for `this.themeToken.theme` too.
Was going to say the same applied for the input group, but there it's a public prop (0.0)